packages/2019-transportation/src/components/SystemWideSummary/SystemWideSummaryVisualization.js

```diff
@@ -0,0 +1,157 @@
+          seconds: result.q1_seconds_late
+        })
+      );
+
+      const medianSeries = data.busSystemWideSummary.value.results.map(
+        result => ({
+          series: "50",
+          arrive_quarter_hour: result.arrive_quarter_hour,
+          seconds: result.median_seconds_late
+        })
+      );
+
+      const thirdQuartileSeries = data.busSystemWideSummary.value.results.map(
+        result => ({
+          series: "75",
+          arrive_quarter_hour: result.arrive_quarter_hour,
+          seconds: result.q3_seconds_late
+        })
+      );
+
+      const ninetyFifthPercentileSeries = data.busSystemWideSummary.value.results.map(
+        result => ({
+          series: "95",
+          arrive_quarter_hour: result.arrive_quarter_hour,
+          seconds: result.p95_seconds_late
+        })
+      );
+
+      setSecondsLateSeries([
+        ...fifthPercentileSeries,
+        ...firstQuartileSeries,
+        ...medianSeries,
+        ...thirdQuartileSeries,
+        ...ninetyFifthPercentileSeries
+      ]);
+    }
+  }, [isLoading]); // eslint-disable-line
+
+  // creating a data series for ons and offs separately and as totals
+  useEffect(() => {
+    if (!isLoading) {
+      const onSeries = data.busSystemWideSummary.value.results.map(result => {
+        return {
+          series: "ons",
+          arrive_quarter_hour: result.arrive_quarter_hour,
+          count: result.total_ons
+        };
+      });
+
+      const offSeries = data.busSystemWideSummary.value.results.map(result => {
+        return {
+          series: "offs",
+          arrive_quarter_hour: result.arrive_quarter_hour,
+          count: result.total_offs
+        };
+      });
+
+      const totalSeries = data.busSystemWideSummary.value.results.map(
+        result => {
+          return {
+            series: "ons_offs",
+            arrive_quarter_hour: result.arrive_quarter_hour,
+            count: result.total_ons + result.total_offs
+          };
+        }
+      );
+
+      setOnsOffsSeries([...onSeries, ...offSeries, ...totalSeries]);
+    }
+  }, [isLoading]); // eslint-disable-line
+
+  return (
+    <>
+      {!isLoading && data && secondsLateSeries && onsOffsSeries ? (
+        <>
+          <LineChart
+            data={onsOffsSeries}
+            dataKey="arrive_quarter_hour"
+            dataSeries="series"
+            dataSeriesLabel={[
+              { category: "ons", label: "Total Ons" },
+              { category: "offs", label: "Total Offs" },
+              { category: "ons_offs", label: "Total Ons and Offs" }
+            ]}
+            dataValue="count"
+            title="System Wide Summary — Ons & Offs"
+            xLabel="Quarter Hour"
+            xNumberFormatter={time => {
+              return time
+                ? `${Math.floor(time)}:${
+                    (time % 1) * 60 === 0 ? "00" : (time % 1) * 60
+                  }`
+                : time;
+            }}
+            yLabel="Ons and Offs"
+          />
```

Comment:
Can you add a comment to say why this is commented out?

Reply:
yes definitely. i had half a thought to add another stacked line chart with the delay time buckets: fifth percentile, quartile, median etc. but i'm not sure it necessarily follows the demo outline that the team is writing.

i'll add that as a note.